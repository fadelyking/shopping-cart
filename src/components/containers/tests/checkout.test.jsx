import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Checkout from "../checkout";
import userEvent from "@testing-library/user-event";
import { describe, expect, vi } from "vitest";

const cart = [
	{ id: 1, name: "shirt", price: 321, image: "link.com" },
	{ id: 2, name: "pants", price: 22, image: "link2.com" },
];

describe("Checkout", () => {
	it("Should render the checkout page", () => {
		const { container } = render(
			<MemoryRouter>
				<Checkout cart={cart} />
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});
});

describe("It should calculate total", () => {
	const cart = [
		{ id: 1, name: "shirt", price: 22.3, image: "link.com" },
		{ id: 2, name: "pants", price: 22.3, image: "link2.com" },
	];
	const { container } = render(
		<MemoryRouter>
			<Checkout cart={cart} />
		</MemoryRouter>
	);

	const total = screen.getByRole("heading", /total: \$0/i);
	expect(total).toBeInTheDocument();
});
