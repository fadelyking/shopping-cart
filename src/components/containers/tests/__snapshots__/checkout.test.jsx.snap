// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkout > Should render the checkout page 1`] = `
<div>
  <div
    class="homeContainer"
  >
    <div
      class="nav-bar"
    >
      <div
        class="nav-links"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              <h1>
                Home
              </h1>
            </a>
          </li>
          <a
            href="/shop"
          >
            <li>
              Shop
            </li>
          </a>
        </ul>
      </div>
      <a
        href="/checkout"
      >
        <div
          class="cart"
        >
          <p>
             Cart Icon
          </p>
        </div>
      </a>
    </div>
    <div
      class="main-checkout"
    >
      <div
        class="right-container"
      >
        <div
          class="cart-item"
        >
          <div
            class="cart-item-left"
          >
            <div
              class="item-price"
            >
              $
              321
            </div>
          </div>
          <div
            class="cart-item-right"
          >
            <div
              class="item-title"
            />
          </div>
        </div>
        <div
          class="cart-item"
        >
          <div
            class="cart-item-left"
          >
            <div
              class="item-price"
            >
              $
              22
            </div>
          </div>
          <div
            class="cart-item-right"
          >
            <div
              class="item-title"
            />
          </div>
        </div>
      </div>
      <div
        class="left-container"
      >
        TOTAL: $
        343
      </div>
    </div>
  </div>
</div>
`;
